src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.cs

```diff
@@ -1727,9 +1727,7 @@ private BoundNode GetBoundLambdaOrQuery(CSharpSyntaxNode lambdaOrQuery)
 
                 // https://github.com/dotnet/roslyn/issues/35038: We need to do a rewrite here, and create a test that can hit this.
 #if DEBUG
-                var diagnostics = new DiagnosticBag();
-                ImmutableDictionary<Symbol, Symbol> ignored = null;
-                _ = RewriteNullableBoundNodesWithSnapshots(boundOuterExpression, incrementalBinder, diagnostics, takeSnapshots: false, snapshotManager: out _, remappedSymbols: ref ignored);
+                AnalyzeBoundNodeNullability(boundOuterExpression, incrementalBinder, diagnostics: new DiagnosticBag(), takeSnapshots: false);
```

Comment:
>AnalyzeBoundNodeNullability(boundOuterExpression, incrementalBinder, diagnostics: new DiagnosticBag(), takeSnapshots: false); [](start = 16, length = 125)

Are we testing this code path given the comment above? Otherwise, consider removing this code and deciding what is the right thing to do while addressing the issue.

Reply:
We're not explicitly hitting it, but there are tests with nullable disabled that will hit this by matter of incidence. I think it's worth keeping this to at least make sure we don't crash in that scenario.

---
In reply to: [441146882](https://github.com/dotnet/roslyn/pull/42403#discussion_r441146882) [](ancestors = 441146882)